lib/ansible/modules/monitoring/zabbix/zabbix_host_facts.py

```diff
@@ -0,0 +1,245 @@
+            listed_hostnames.append(zabbix_host['name'])
+        return unique_hosts
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(type='str', required=True, aliases=['url']),
+            login_user=dict(type='str', required=True),
+            login_password=dict(type='str', required=True, no_log=True),
+            host_name=dict(type='str', default='', required=False),
+            host_ip=dict(type='list', default=[], required=False),
+            http_login_user=dict(type='str', required=False, default=None),
+            http_login_password=dict(type='str', required=False, default=None, no_log=True),
+            timeout=dict(type='int', default=10),
+            exact_match=dict(type='bool', required=False, default=False),
+            remove_duplicate=dict(type='bool', required=False, default=False)
+        ),
+        supports_check_mode=True
+    )
+
+    if not HAS_ZABBIX_API:
+        module.fail_json(msg="Missing required zabbix-api module (check docs or install with: pip install zabbix-api)")
+
+    server_url = module.params['server_url']
+    login_user = module.params['login_user']
+    login_password = module.params['login_password']
+    http_login_user = module.params['http_login_user']
+    http_login_password = module.params['http_login_password']
+    host_name = module.params['host_name']
+    host_ips = module.params['host_ip']
+    timeout = module.params['timeout']
+    exact_match = module.params['exact_match']
+    is_remove_duplicate = module.params['remove_duplicate']
+
+    zbx = None
+    # login to zabbix
+    try:
+        zbx = ZabbixAPIExtends(server_url, timeout=timeout, user=http_login_user, passwd=http_login_password)
+        zbx.login(login_user, login_password)
+    except Exception as e:
+        module.fail_json(msg="Failed to connect to Zabbix server: %s" % e)
+
+    host = Host(module, zbx)
+
+    if host_name:
+        is_host_exist = host.is_host_exist(host_name, exact_match)
+
+        if is_host_exist:
+            hosts = host.get_hosts_by_host_name(host_name, exact_match)
+            if is_remove_duplicate:
+                hosts = host.delete_duplicate_hosts(hosts)
+            extended_hosts = []
+            for zabbix_host in hosts:
+                zabbix_host['hostinterfaces'] = host._zapi.hostinterface.get({
+                    'output': 'extend', 'hostids': zabbix_host['hostid']
+                })
+                extended_hosts.append(zabbix_host)
+            module.exit_json(ok=True, hosts=extended_hosts)
+        else:
+            module.exit_json(ok=False, hosts=[], result="No Host present")
+    elif host_ips:
+        extended_hosts = host.get_hosts_by_ip(host_ips)
+        if is_remove_duplicate:
+            hosts = host.delete_duplicate_hosts(extended_hosts)
+        module.exit_json(ok=True, hosts=extended_hosts)
+    else:
+        module.exit_json(ok=False, hosts=[], result="No Host present")
```

Comment:
This one is a bit odd as you are expecting that user will skip both `host_name` and `host_ip` arguments. In that case it is equal to not using your module at all. I would just replace with `fail_json`:

```yaml
module.fail_json(msg="No host name or IP provided")
```

Same goes for the line 235, but it should be removed with the whole `is_host_exist` check.